### PR TITLE
feat: remove dark blue theme

### PR DESCRIPTION
**Pull request by zxt-tzx**
saw this issue https://github.com/coder/coder/issues/14811, looks like an easy change

considered trying to modify the type exported from `UpdateUserAppearanceSettingsRequest` in `typesGenerated.ts` to be `"light" | "dark"`, but not super sure how to generate types from the Go codebase, so have left this as it is

### CODE REVIEW

**Comment by zxt-tzx** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by zxt-tzx** [comment]
the [`test-e2e` test](https://github.com/coder/coder/actions/runs/11123247288/job/30915802089?pr=14890) failed, can't tell if it's just a flaky test or something with the changes I've made

```
[stderr] [WebServer] 2024-10-01 13:26:12.993 [info]  coderd.agentrpc.yamux.stdlib: [ERR] yamux: Failed to read header: failed to get reader: failed to read frame header: EOF  owner=admin  workspace_name=43111e0b  agent_name=dev  request_id=108bea26-2d26-46a3-9a22-c675f4e09fa4
```

**Review by aslilac — COMMENTED** [review]
we'll also want to add a db migration that sets the theme of any users currently using the dark blue theme back to the default, otherwise they won't see anything selected on the appearance settings page.

it should just look like:

```sql
UPDATE users SET theme_preference = '' WHERE theme_preference = 'darkBlue';
```

and the down migration can just be an empty file with a comment saying something like

```sql
-- Nothing to restore
```

You can create the migration files by running...

```sh
./coderd/database/migrations/create_migration.sh remove-dark-blue-theme
```

File: site/src/theme/index.ts [hunk 1]
```diff
@@ -33,8 +32,7 @@ export const DEFAULT_THEME = "dark";
 
 const theme = {
 	dark,
-	darkBlue,
 	light,
-} satisfies Record<string, Theme>;
+} satisfies Record<"dark" | "light", Theme>;
```
[the remark this answers is not in the document]
Comment (zxt-tzx) [reply site/src/theme/index.ts:36]: happy to restore it!

**Review by aslilac — APPROVED** [review]
looks great! thank you!

**Pull request merged**